pkg/apis/kops/validation/validation.go

```diff
@@ -1085,6 +1085,10 @@ func validateNetworkingCalico(v *kops.CalicoNetworkingSpec, e kops.EtcdClusterSp
 		allErrs = append(allErrs, IsValidValue(fldPath.Child("awsSrcDstCheck"), &v.AWSSrcDstCheck, valid)...)
 	}
 
+	if v.CrossSubnet != nil {
+		field.Invalid(fldPath.Child("crossSubnet"), v.CrossSubnet, "crossSubnet is deprecated, use awsSrcDstCheck instead")
```

Comment:
As coded here, It's not just deprecated.

I think we shouldn't validate this. At most, only fail validation if `CrossSubnet` is true and `AWSSrcDstCheck` is `DoNothing`.

Reply:
This was semi deprecated in docs a while ago. It shouldn't be used since kOps 1.19.
We should make it very clear that something i not being used anymore instead of just moving on.

Reply:
If you have a strong preference here, I don't mind doing it though.

Reply:
We're effectively making it default to `true`. So we shouldn't make it fail validation if it explicitly requests the now-default behavior.

It wasn't possible to explicitly turn off since it wasn't previously a pointer. So the only thing where it makes sense to fail validation is if it was explicitly set and something else contradictory is set, for example `AWSSrcDstCheck` set to `DoNothing`.

Reply:
OK, updated.